src/openrct2/ride/Ride.cpp

```diff
@@ -558,6 +558,8 @@ bool track_block_get_next(CoordsXYE* input, CoordsXYE* output, int32_t* z, int32
         return false;
 
     trackBlock += inputElement->GetSequenceIndex();
+    if (trackBlock == nullptr)
+        return false;
```

Comment:
I think this is mostly luck. If the sequence index is higher then it could be a non-nullptr but still invalid pointer. Can we instead check if the sequence index is greater than the number of sequences that track block has?

Reply:
Does it have to be "instead", or can I do both?